app/models/teacher.rb

```diff
@@ -45,4 +46,28 @@ def send_magic_link(*)
   def send_devise_notification(notification, *args)
     devise_mailer.send(notification, self, *args).deliver_later
   end
+
+  def should_send_reminder_email?
+    all_reference_requests =
+      ReferenceRequest
+        .joins(:work_history)
+        .where(work_histories: { application_form_id: application_form.id })
+        .where(received_at: nil, expired_at: nil)
+    @remindable_reference_requests =
+      all_reference_requests.select do |reference_request|
+        reference_request.should_send_reminder_email?(
+          (reference_request.expires_at.to_date - Time.zone.today).to_i,
+          reminder_emails.count,
+        )
+      end
+    Rails.logger.debug @remindable_reference_requests
+    @remindable_reference_requests.any?
+  end
+
+  def send_reminder_email(_number_of_reminders_sent)
```

Comment:
We've got `number_of_reminders_sent` here available so we can pass it as a parameter and then use it in the view, this avoids issues around delays in email jobs and the database changing.

Reply:
```rb
TeacherMailer
      .with(teacher: self, reference_requests: @remindable_reference_requests, number_of_reminders_sent:)
```

and then in the mailer:

```rb
@number_of_reminders_sent = params[:number_of_reminders_sent]
```